components/form/CheckboxGroup.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { useFormContext } from 'react-hook-form';
+import PropTypes from 'prop-types';
+import { Label, ErrorLabel } from '.';
+
+export function Checkbox({ containerClassName, inputClassName, text, name, value }) {
+  const { register } = useFormContext();
+  return (
+    <>
+      <div className={`flex items-center ${containerClassName}`}>
+        <input
+          id={`checkbox-${name}-${value.toString()}`}
+          className={`w-4 h-4 text-amcovad-primary-400 border-gray-200 rounded focus:ring-amcovad-primary-300 ${inputClassName} `}
+          name={name}
+          type="checkbox"
+          {...register(name)}
+          value={value}
+        />
+        <Label name={name} htmlFor={`checkbox-${name}-${value.toString()}`} text={text} />
```

Comment:
Now if we use label above, then we can use text here. It is not compulsory that the props name and props value should have the same name.

```suggestion
        <Label name={name} htmlFor={`checkbox-${name}-${value.toString()}`} text={label} />
```